src/rust/id.shinigami/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+    let url = format!(
+        "{}/v1/chapter/{}/list?page=1&page_size=3000&sort_by=chapter_number&sort_order=desc",
+        API_URL, id
+    );
+    let json = Request::get(url)
+        .header("Accept", "application/json")
+        .header("Origin", BASE_URL)
+        .json()?;
+
+    let data = json.as_object()?;
+    if data.get("retcode").as_int()? != 0 {
+        return Ok(Vec::new());
+    }
+
+    let chapter_list = data.get("data").as_array()?;
+    let mut chapters = Vec::new();
+
+    for chapter in chapter_list {
+        let chapter_obj = chapter.as_object()?;
+        let chapter_number = chapter_obj.get("chapter_number").as_float()? as f32;
+        let chapter_id = chapter_obj.get("chapter_id").as_string()?.read();
+        
+        chapters.push(Chapter {
+            id: chapter_id.clone(),
+            title: format!(
+                "Chapter {}", 
+                chapter_number.to_string().replace(".0", "")
+            ),
+            chapter: chapter_number,
+            url: format!("{}/chapter/{}", BASE_URL, chapter_id),
+            date_updated: helpers::parse_date(chapter_obj.get("release_date").as_string()?.read()),
+            ..Default::default()
+        });
+    }
+
+    Ok(chapters)
+}
+
+#[get_manga_details]
+fn get_manga_details(id: String) -> Result<Manga> {
+    let url = format!("{}/v1/manga/detail/{}", API_URL, id);
+    let json = Request::get(url)
+        .header("Accept", "application/json")
+        .header("Origin", BASE_URL)
+        .json()?;
+
+    let data = json.as_object()?;
+    if data.get("retcode").as_int()? != 0 {
+        return Err(aidoku::error::AidokuError { 
+            reason: aidoku::error::AidokuErrorKind::NodeError(aidoku::error::NodeError::ParseError) 
+        });
+    }
+
+    let manga_obj = data.get("data").as_object()?;
+    let manga_id = manga_obj.get("manga_id").as_string()?.read();
+    
+    let cover = if let Ok(url) = manga_obj.get("cover_portrait_url").as_string() {
+        if !url.read().is_empty() {
+            url.read()
+        } else {
+            manga_obj.get("cover_image_url").as_string()?.read()
+        }
+    } else {
+        manga_obj.get("cover_image_url").as_string()?.read()
+    };
+
+    let taxonomy = manga_obj.get("taxonomy").as_object()?;
+    let mut genres = Vec::new();
+    if let Ok(genre_array) = taxonomy.get("Genre").as_array() {
+        for genre in genre_array {
+            if let Ok(genre_obj) = genre.as_object() {
+                if let Ok(genre_name) = genre_obj.get("name").as_string() {
+                    genres.push(genre_name.read());
+                }
+            }
+        }
+    }
+
+    Ok(Manga {
+        id: manga_id.clone(),
+        title: manga_obj.get("title").as_string()?.read(),
+        author: helpers::get_taxonomy_names(&taxonomy, "Author"),
+        artist: helpers::get_taxonomy_names(&taxonomy, "Artist"),
+        description: manga_obj.get("description").as_string()?.read(),
+        url: manga_id,
```

Comment:
this isn't a valid url (same in get_manga_list)

Reply:
i fix it to url: format!("{}/manga/{}", BASE_URL, manga_id),
is it okay?

Reply:
doesn't look like those are the correct links. you should test by pressing the safari icon on a manga page.

Reply:
okay, fixed